src/chunk.c

```diff
@@ -1016,6 +1173,158 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
+			newer_than_type != InvalidOid &&
+			older_than_type != newer_than_type)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("older_than_type and newer_than_type should have the same type")
+					 ));
+
+		/*
+		 * Cache outside the if block to make sure cached hypertable entry
+		 * returned will still be valid in foreach block below
+		 */
+		hypertable_cache = hypertable_cache_pin();
+		if (PG_ARGISNULL(0))
+		{
+			hypertables = hypertable_get_all();
+		}
+		else
+		{
+			ht = hypertable_cache_get_entry(hypertable_cache, table_relid);
+			if (!ht)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("The given table does not exist or is not a hypertable")
+						 ));
+			hypertables = list_make1(ht);
+		}
+
+		foreach(lc, hypertables)
+		{
+			ht = lfirst(lc);
+
+			time_dim = hyperspace_get_open_dimension(ht->space, 0);
+
+			if (time_dim_type == InvalidOid)
+				time_dim_type = time_dim->fd.column_type;
+
+			/*
+			 * Even though internally all time columns are represented as
+			 * bigints, it is locally unclear what set of chunks should be
+			 * returned if there are multiple tables on the system some of
+			 * which care about timestamp when others do not. That is why,
+			 * whenever there is any time dimension constraint given as an
+			 * argument (older_than or newer_than) we make sure all
+			 * hypertables have the time dimension type of the given type or
+			 * through an error. This check is done accross hypertables that
+			 * is why it is not in the helper function below.
+			 */
+			if (time_dim_type != time_dim->fd.column_type &&
+				(older_than_type != InvalidOid || newer_than_type != InvalidOid))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot call \"%s\" on all hypertables "
+								"when all hypertables do not have the same time dimension type",
+								caller_name)
+						 ));
+
+			chunks_added = chunks_find_all_in_range_limit_typecheck_wrapper(ht->space,
+																			time_dim,
+																			older_than_datum,
+																			older_than_type,
+																			newer_than_datum,
+																			newer_than_type,
+																			-1,
+																			funcctx->multi_call_memory_ctx,
+																			caller_name);
+
+			if (NULL == chunks)
+			{
+				chunks = chunks_added;
+			}
+			else
+			{
+				/*
+				 * chunks_addall uses repalloc and this switch makes sure the
+				 * additional memory is allocated in the correct location
+				 */
+				MemoryContext oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+
+				chunks = chunks_addall(chunks, chunks_added);
+				MemoryContextSwitchTo(oldcontext);
+			}
+		}
+		cache_release(hypertable_cache);
+
+		/* There are no hypertables in the database so return 0 rows */
+		if (NULL == chunks)
+		{
+			MemoryContext oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+
+			chunks = chunks_alloc(0);
```

Comment:
I just added this. Previously, client crashed if there were no hypertables and `show_chunks` was called. 
As an alternative, it would make sense to error in here instead of returning 0 rows. That is what, for example `\dt` does in psql. Let me know what you think. @cevian